app/shared/types/templateSchema.ts

```diff
@@ -155,12 +155,79 @@ ajv.addKeyword('cantDeleteInheritedProperties', {
   },
 });
 
+ajv.addKeyword('cantReuseNameWithDifferentType', {
+  async: true,
+  errors: true,
+  type: 'object',
+  async validate(_schema: any, template: TemplateSchema) {
+    if (!template.properties || template.properties.length === 0) {
+      return true;
+    }
+
+    const condition = template.properties.map(property => ({
+      $and: [
+        {
+          name: property.name,
+          $or: [
+            { content: { $ne: property.content } },
+            { type: { $ne: property.type } },
+            { relationtype: { $ne: property.relationType } },
+          ],
+        },
+      ],
+    }));
+    const query = {
+      $and: [
+        { _id: { $ne: template._id } },
+        { properties: { $elemMatch: { $or: [...condition] } } },
+      ],
+    };
+    const matchedTemplates = await model.get(query);
+    if (matchedTemplates.length > 0) {
+      const allProperties: PropertySchema[] = matchedTemplates.reduce(
+        (memo: PropertySchema[], t) => memo.concat(t.properties || []),
+        []
+      );
+
+      const errorProperties = template.properties.reduce((propertyNames: string[], property) => {
+        const matches = allProperties.find(
+          p =>
+            p.name === property.name &&
+            (p.content !== property.content ||
+              p.type !== property.type ||
+              p.relationtype !== property.relationType)
+        );
+
+        if (matches && !propertyNames.includes(property?.name || '')) {
+          return propertyNames.concat([property?.name || '']);
+        }
```

Comment:
property name in this scenario should be always there, defaulting to blank could hide errors, i recomend in this scenarios to use the ensure utility function (from app/shared/tsUtils) until we figure out or work more on our base TS typings.
should be something like this.
```
ensure(property.name)

Reply:
Great! thanks for this tip @daneryl, it's very usefull